### PR TITLE
Set up backend

### LINKED ISSUE #2: Set up initial backend repo
### Description
Need to set up backend repository before the backend team starts working on any code. What we aim for is a properly structured backend codebase that is easy to maintain and leverages best practices.

### Acceptance Criteria
- [x] Set up typescript and ts-node
- [x] Creating basic model/route/controller
- [x] Set up app.ts (cors and stuff)
- [x] Link to mongodb
- [x] Add types to parameters whenever possible
- [x] Mongoose model to link to interface (for autosuggest)
- [x] Add middleware for handling errors, and add logger
- [x] Structure code into model, route, controller
- [x] Remove require module imports
- ESLint <= to be done through #15 

### Implementation details
- Set up typescript node & express & mongodb
- Used npm for dependency management (so make sure to only use npm)
    - `npm i` then `npm start` to run the backend
- Used nested routing (express) to set up sample endpoints (refer to 750 wk3 modules for these)
- .env file should be placed under backend folder, and have MONGODB_URL, PORT
- Project structure:
    - src
        - models: has all mongodb database schema (eg person.model.ts)
        - routes: configures REST routes (eg person.route.ts)
        - controllers: defines async functions used by routes (eg person.controller.ts)
        - utils: configures environment values, middleware for logging and catching errors
        - index.ts: entrypoint that configures mongoose & express
 
### Testing notes
n/a

### LINKED ISSUE #15: ESLint for backend
### Description

ESLint should be configured so that coding standard is enforced. Do with #14 

### Acceptance Criteria
- [x] Add ESLint to enforce codestyle
- [x] Add run-scripts for lint and lint-fix

### Implementation
Extends AirBnB guideline
Can run lint check through `npm run lint`
Automatic lint fix through `npm run lint-fix`

### Testing Information
n/a

### PR DESCRIPTION
**Pull request by hajineats**
- Set up typescript node & express & mongodb
- Used npm for dependency management (so make sure to only use npm)
    - `npm i` then `npm start` to run the backend
- Used nested routing (express) to set up sample endpoints (refer to 750 wk3 modules for these)
- .env file should be placed under backend folder, and have MONGODB_URL, PORT
- Project structure:
    - src
        - models: has all mongodb database schema (eg person.model.ts)
        - routes: configures REST routes (eg person.route.ts)
        - controllers: defines async functions used by routes (eg person.controller.ts)
        - utils: configures environment values, middleware for logging and catching errors
        - index.ts: entrypoint that configures mongoose & express
- ESLint added for code style enforcing. Extended AirBnB style
    - Run lint through `npm run lint`, and fix lint issues through `npm run lint-fix`

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,134 @@
+node_modules
+.env
```
Comment (butterplup) [inline backend/.gitignore:2]: .env is already mentioned on line 80

Comment (butterplup) [reply backend/.gitignore:2]: and it seems node_modules is also already present on line 46

**Review by butterplup — CHANGES_REQUESTED** [review]
Would you be able to add a newline character at the end of each file so that it meets the requirements for ESLint?

File: backend/.gitignore (hunk 1 above)
Comment (hajineats) [reply backend/.gitignore]: fair observation, thanks!

**Comment by hajineats** [comment]
> Would you be able to add a newline character at the end of each file so that it meets the requirements for ESLint?

Yep done. It would probably be a good idea to leave any other formatting issues until eslint is set up for backend (#15)

**Review by butterplup — APPROVED** [review]
LGTM!

**Review by sydive — APPROVED** [review]
LGTM 👍 
🐜 We should probably change the folder name to `forgettable-backend` from `backend` at some point, but otherwise looks good!

**Pull request merged**